src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private int nodeIndex;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<T>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " is out of list");
+        }
+        if (size == index) {
+            add(value);
+        } else {
+            Node<T> currentNode = getNode(index);
+            Node<T> addedNode = new Node<>(currentNode.prev, value, currentNode);
```

Comment:
It's better to slightly change order of actions and reuse checkIndex method:
```suggestion
        if (size == index) {
            add(value);
            return;
        } 
        checkIndex(index);
        Node<T> currentNode = getNode(index);
        Node<T> addedNode = new Node<>(currentNode.prev, value, currentNode);
```